crates/ruff_python_formatter/src/statement/stmt_class_def.rs

```diff
@@ -1,12 +1,133 @@
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::StmtClassDef;
+use crate::comments::trailing_comments;
+use crate::expression::parentheses::Parenthesize;
+use crate::prelude::*;
+use crate::trivia::{first_non_trivia_token, SimpleTokenizer, Token, TokenKind};
+use crate::USE_MAGIC_TRAILING_COMMA;
+use ruff_formatter::{format_args, write};
+use ruff_text_size::TextRange;
+use rustpython_parser::ast::{Expr, Keyword, Ranged, StmtClassDef};
 
 #[derive(Default)]
 pub struct FormatStmtClassDef;
 
 impl FormatNodeRule<StmtClassDef> for FormatStmtClassDef {
     fn fmt_fields(&self, item: &StmtClassDef, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtClassDef {
+            range: _,
+            name,
+            bases,
+            keywords,
+            body,
+            decorator_list,
+        } = item;
+
+        f.join_with(hard_line_break())
+            .entries(decorator_list.iter().formatted())
+            .finish()?;
+
+        if !decorator_list.is_empty() {
+            hard_line_break().fmt(f)?;
+        }
+
+        write!(f, [text("class"), space(), name.format()])?;
+
+        if !(bases.is_empty() && keywords.is_empty()) {
+            write!(
+                f,
+                [group(&format_args![
+                    text("("),
+                    soft_block_indent(&FormatInheritanceClause {
+                        class_definition: item
+                    }),
+                    text(")")
+                ])]
+            )?;
+        }
+
+        let comments = f.context().comments().clone();
+        let trailing_head_comments = comments.dangling_comments(item);
+
+        write!(
+            f,
+            [
+                text(":"),
+                trailing_comments(trailing_head_comments),
+                block_indent(&body.format())
+            ]
+        )
+    }
+
+    fn fmt_dangling_comments(
+        &self,
+        _node: &StmtClassDef,
+        _f: &mut PyFormatter,
+    ) -> FormatResult<()> {
+        // handled in fmt_fields
+        Ok(())
+    }
+}
+
+struct FormatInheritanceClause<'a> {
+    class_definition: &'a StmtClassDef,
+}
+
+impl Format<PyFormatContext<'_>> for FormatInheritanceClause<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        let StmtClassDef {
+            bases,
+            keywords,
+            name,
+            ..
+        } = self.class_definition;
+
+        let separator = format_with(|f| write!(f, [text(","), soft_line_break_or_space()]));
+        let source = f.context().contents();
+
+        let mut joiner = f.join_with(&separator);
+
+        if let [first, rest @ ..] = bases.as_slice() {
+            // Manually handle parentheses here because the expression is parenthesized incorrectly assumes that the
+            // parentheses from the inheritance clause belong to the expression.
```

Comment:
don't understand that comment

Reply:
I reworded the comment and added an example.